### ISSUE TITLE
Add option to include inferred types in generated stubs

### ISSUE THREAD

**Issue by codebreach** [microsoft/pyright#1916]
**Is your feature request related to a problem? Please describe.**
Stubs generated do not have any types, however inference is able to detect types. Would be useful to have these types available in generated stubs

**Describe the solution you'd like**
Have an option to infer types for stubs `pyright --createstubs foo --infer`

**Additional context**

**Comment by erictraut** [comment]
When I initially implemented the stub generation logic, I explored the idea of including inferred types for function return values, but I found that it generated inaccurate types too often. It resulted in more work rather than less to clean it up. I decided that it was better to omit return type annotations rather than infer them. At least it is then obvious which return type annotations were missing in the original source code.

So, while I like this suggestion in concept, I don't think it works in practice.

**Comment by erictraut** [comment]
Here's a proposal that I think makes for a good compromise. I could emit a comment that includes the inferred return type. When you're "cleaning up" the stub, you can choose to delete or replace the comment if you disagree with the inferred type or you can uncomment it if you agree with it.

For example, 

```python
class Foo:
    def is_finished(self): # -> bool:
        ...

    def get_message_details(self): # -> List[str]:
        ...
```

What do you think?

**Comment by erictraut** [comment]
I've implemented this mechanism, and I'm pretty happy with the results. This will be included in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is included in pyright 1.1.145, which I just published.

**Comment by codebreach** [comment]
This is awesome and should really help with the stub creation process!

Thanks for the quick turnaround @erictraut

Will take it for a spin and let you know if there are any issues.
